scripts/obwatch/ob-watcher.py

```diff
@@ -235,18 +150,129 @@ def create_orderbook_obj(self):
+            plt.hist(orderfees, bins, rwidth=0.8, range=(0, orderfees[0] * 2))
+        else:
+            plt.hist(orderfees, bins, rwidth=0.8)
+        plt.grid()
+        plt.title('CoinJoin Orderbook Depth Chart for amount=' + str(cj_amount /
+                                                                     1e8) + 'btc')
+        plt.xlabel('CoinJoin Fee / btc')
+        plt.ylabel('Frequency')
+        return get_graph_html(fig)
+
+    def create_size_histogram(self, args):
+        try:
+            self.taker.dblock.acquire(True)
+            rows = self.taker.db.execute('SELECT maxsize, ordertype FROM orderbook;').fetchall()
+        finally:
+            self.taker.dblock.release()
+        rows = [o for o in rows if o["ordertype"] in filtered_offername_list]
+        ordersizes = sorted([r['maxsize'] / 1e8 for r in rows])
+
+        fig = plt.figure()
+        scale = args.get("scale")
+        if (scale is not None) and (scale[0] == "log"):
+            ratio = ordersizes[-1] / ordersizes[0]
+            step = ratio ** 0.0333  # 1/30
+            bins = [ordersizes[0] * (step ** i) for i in range(30)]
+        else:
+            bins = 30
+        plt.hist(ordersizes, bins, histtype='bar', rwidth=0.8)
+        if bins is not 30:
+            fig.axes[0].set_xscale('log')
+        plt.grid()
+        plt.xlabel('Order sizes / btc')
+        plt.ylabel('Frequency')
+        return get_graph_html(fig) + ("<br/><a href='?scale=log'>log scale</a>" if
+                                      bins == 30 else "<br/><a href='?'>linear</a>")
+
+    def create_orderbook_table(self, btc_unit, rel_unit):
+        result = ''
+        try:
+            self.taker.dblock.acquire(True)
+            rows = self.taker.db.execute('SELECT * FROM orderbook;').fetchall()
+        finally:
+            self.taker.dblock.release()
+        if not rows:
+            return 0, result
+        #print("len rows before filter: " + str(len(rows)))
+        rows = [o for o in rows if o["ordertype"] in filtered_offername_list]
+        order_keys_display = (('ordertype', ordertype_display),
+                              ('counterparty', do_nothing), ('oid', order_str),
+                              ('cjfee', cjfee_display), ('txfee', satoshi_to_unit),
+                              ('minsize', satoshi_to_unit),
+                              ('maxsize', satoshi_to_unit))
+
+        # somewhat complex sorting to sort by cjfee but with swabsoffers on top
+
+        def orderby_cmp(x, y):
+            def findindex(list, item):
+                for i, node in enumerate(list):
+                    if node == item:
+                        return i
+                return None
+            if x['ordertype'] == y['ordertype']:
+                return cmp(Decimal(x['cjfee']), Decimal(y['cjfee']))
+            return cmp(findindex(offername_list, x['ordertype']),
+                       findindex(offername_list, y['ordertype']))
```

Comment:
`offername_list` should be made to a list in `jmdaemon/protocol.py` and this workaround be removed